### DISCUSSION TITLE
引言&思考 | Alex Lin's Notes

### DISCUSSION THREAD

**Discussion by giscus** [chaoskyme/chaoskyme.github.io#324] [Comment]
# 引言&思考 | Alex Lin's Notes

2020-06-04多用库，少用框架库和框架的一个区别是，一次可以使用多个库，但是框架一次只能使用一个。 这就是为什么我不使用框架的原因。一旦用了一个框架，就无法再用另一个框架了，移植的成本太高了。使用库就没有这个问题。 2020-05-24程序员的Johnston规则 保持清醒的头脑（不要急躁） 充分休息后再工作（劳累时不要编写程序） 程序应简单、优美（Kiss，即Keep It Simpl

https://chaosky.tech/quotes/

**Comment by xwal** [comment]
## 2022-11-24
The best way to learn is by teaching.

**Comment by xwal** [comment]
## 2022-11-30

天命不足畏，祖宗不足法，人言不足恤。

**Comment by xwal** [comment]
## 2022-12-02

<img width="600" alt="iShot_2022-12-02_14 06 22" src="https://user-images.githubusercontent.com/1863204/205226582-884888f6-3a7c-499a-ac8a-11764b098b8c.png">

The specific types of hash functions we are going to focus on are **cryptographic** hash functions. These hash functions need five specific properties. They must be:

- 🔮 **Deterministic** - One specific input always maps to the **same** specific output
- 🎲 **Pseudorandom** - It is not possible to guess the output based on the output of similar inputs
- ➡️ **One-way** - If someone gives you a new output, you could not determine an input without guessing
- 🏎️ **Fast to Compute** - It must be a quick calculation for a computer
- 💥 **Collision-resistant** - The chance of a collision should be infinitesimally small

**Comment by xwal** [comment]
## 2022-12-06

### Public Key Cryptography

[Slides](https://docs.google.com/presentation/d/e/2PACX-1vSUqk6Uvnole2hTr9r7-UBcvmPquLhMNa_qPHL26BoVCk0v5j2EvsY2UyO6n2JbB7OfxJVArXlBPiAG/pub?start=false&loop=false&delayms=3000)


<img width="500" alt="iShot_2022-12-06_09 56 10" src="https://user-images.githubusercontent.com/1863204/205794972-440440a3-4615-441c-965b-bd8d8cf5b19a.png">

<img width="500" alt="iShot_2022-12-06_09 56 29" src="https://user-images.githubusercontent.com/1863204/205795007-f3b7fe5e-7d6b-4f58-9826-50825dbf4da3.png">
<img width="500" alt="iShot_2022-12-06_09 56 58" src="https://user-images.githubusercontent.com/1863204/205795043-41753071-9ee1-465a-9ea4-f14d2d244b60.png">
<img width="500" alt="iShot_2022-12-06_10 35 49" src="https://user-images.githubusercontent.com/1863204/205795050-f848af50-4745-4802-bcb4-d396a447654e.png">
<img width="500" alt="iShot_2022-12-06_10 36 03" src="https://user-images.githubusercontent.com/1863204/205795054-9204f07e-64d8-4598-b546-14c2ad57211b.png">

**Comment by xwal** [comment]
## 2023-01-26

### Patricia Merkle Tries
#### Bitcoin: Block Architecture
![image](https://user-images.githubusercontent.com/1863204/214904588-8b3bc5cb-0a02-49d1-ae48-d3d19fb684c2.png)

#### Ethereum Block Architecture
![image](https://user-images.githubusercontent.com/1863204/214904749-364eb641-bb0b-4397-a70c-3051c02a71fd.png)

**Comment by xwal** [comment]
## 2023-11-11

### The *OS Architecture
![](https://github.com/xwal/xwal.github.io/assets/1863204/9920c9f1-478c-4afa-a680-4bcca9b955a6)

**Comment by xwal** [comment]
## 2023-11-28

这世上的事儿，凡是禁止的，往往是有好处但不想分给你。凡是提倡的，大概率是有坑需要你去填。

**Comment by xwal** [comment]
## 2024-01-14
![What is OSI model](https://github.com/xwal/xwal.github.io/assets/1863204/373b013f-81a8-4531-9ac4-cc81172065b7)

![Logging, Tracing, Metrics](https://github.com/xwal/xwal.github.io/assets/1863204/922d1df2-f895-4bf7-b728-634eb3198a8c)

## 2024-01-31
![IMG_7339](https://github.com/xwal/xwal.github.io/assets/1863204/b73e7603-850e-46f5-8a83-46cde75659f6)

**Reply by xwal** [reply]
First do it, then do it right, then do it better.

**Comment by xwal** [comment]
## 2024-02-15

止损决定了交易寿命，止盈决定了一年别人赚10倍而你只能赚2倍。

**Comment by xwal** [comment]
## 2024-08-11
![ed998b2e-fbc8-4c3c-b339-eca5abd85ce3_1289x1536](https://github.com/user-attachments/assets/2450aa04-94ac-4fdc-beea-127d4e9e2003)
SSH has two versions: SSH-1 and SSH-2. SSH-2 was standardized by the IETF.
It has three main layers: Transport Layer, Authentication Layer, and Connection Layer.

Transport Layer
The Transport Layer provides encryption, integrity, and data protection to ensure secure communication between the client and server.

Authentication Layer
The Authentication Layer verifies the identity of the client to ensure that only authorized users can access the server.

Connection Layer
The Connection Layer multiplexes the encrypted and authenticated communication into multiple logical channels.

**Reply by xwal** [reply]
### 2024-08-28
欲买桂花同载酒，终不似，少年游。

**Comment by xwal** [comment]
## 2024-12-07
评估自己的能力“用乘法而不是加法”，可能初看起来不太容易理解，但也很好懂。 人的能力通常是参差不齐的，有长处就必然有短处。要想成为好的技术人员，技术是不能 “一俊遮百丑” 的。以 1 分为满分，如果你的技术有 0.8 分，表达能力只有 0. 1 分，总分可能不是（0.8+0.1)/2=0.45 分，而是 0.8x0.1 =0.08 分。如果给你时间把自己的技能提升 0.1 分，总分可以到 0.09 分，而用来提高表达能力，则可以达到 0.16 分。哪种选择对个人更有利，其实是一目了然的。不幸的是，大家通常在潜意识里更愿意遮挡自己的短处，更习惯于训练提高自己的长处。程序员更是如此，面对非技术问题，他们往往更希望从技术方面解决。 仅仅是因为他们更“喜欢”从这方面人手，而没有考虑这样做是否真的有效率。想要成为优秀的技术人员，一定要能克服情感的抵触，注重补齐短板。

**Comment by xwal** [comment]
## 2025-02-07

定义一个资产就三个维度：稀缺性，收益性，流动性。

**Comment by xwal** [comment]
## 2025-03-25
講，你又唔聽；聽，你又唔明；明，你又唔做；做，你又做錯；錯，你又唔認；認，你又唔改；改，你又唔服；唔服又唔講！

**Comment by xwal** [comment]
## 2025-04-06 破窑赋

　　天有不测风云，人有旦夕祸福。蜈蚣百足，行不及蛇；雄鸡两翼，飞不过鸦。马有千里之程，无骑不能自往；人有冲天之志，非运不能自通。

　　盖闻：人生在世，富贵不能淫，贫贱不能移。文章盖世，孔子厄于陈邦；武略超群，太公钓于渭水。颜渊命短，殊非凶恶之徒；盗跖年长，岂是善良之辈。尧帝明圣，却生不肖之儿；瞽叟愚顽，反生大孝之子。张良原是布衣，萧何称谓县吏。晏子身无五尺，封作齐国宰相；孔明卧居草庐，能作蜀汉军师。楚霸虽雄，败于乌江自刎；汉王虽弱，竟有万里江山。李广有射虎之威，到老无封；冯唐有乘龙之才，一生不遇。韩信未遇之时，无一日三餐，及至遇行，腰悬三尺玉印，一旦时衰，死于阴人之手。

　　有先贫而后富，有老壮而少衰。满腹文章，白发竟然不中；才疏学浅，少年及第登科。深院宫娥，运退反为妓妾；风流妓女，时来配作夫人。

　　青春美女，却招愚蠢之夫；俊秀郎君，反配粗丑之妇。蛟龙未遇，潜水于鱼鳖之间；君子失时，拱手于小人之下。衣服虽破，常存仪礼之容；面带忧愁，每抱怀安之量。时遭不遇，只宜安贫守份；心若不欺，必然扬眉吐气。初贫君子，天然骨骼生成；乍富小人，不脱贫寒肌体。

　　天不得时，日月无光；地不得时，草木不生；水不得时，风浪不平；人不得时，利运不通。注福注禄，命里已安排定，富贵谁不欲？人若不依根基八字，岂能为卿为相？

　　吾昔寓居洛阳，朝求僧餐，暮宿破窑，思衣不可遮其体，思食不可济其饥，上人憎，下人厌，人道我贱，非我不弃也。今居朝堂，官至极品，位置三公，身虽鞠躬于一人之下，而列职于千万人之上，有挞百僚之杖，有斩鄙吝之剑，思衣而有罗锦千箱，思食而有珍馐百味，出则壮士执鞭，入则佳人捧觞，上人宠，下人拥。人道我贵，非我之能也，此乃时也、运也、命也。

　　嗟呼！人生在世，富贵不可尽用，贫贱不可自欺，听由天地循环，周而复始焉。

**Comment by xwal** [comment]
## 2025-04-07 青云志

身处低谷不自弃，我命由我不由天。
无人扶我青云志，我自踏雪至山巅。
若是命中无此运，亦可孤身登昆仑。
红尘赠我三尺剑，酒看瘦马一世街。
世人朝路乃绝涧，独见众生止步前。
海到尽头天作岸，山登绝顶我为峰。
如若东山能再起，大鹏展翅九万里。
一入红尘梦易真，一朝悟透心境名。
一朝悟道见真我，昔日枷锁皆云烟。
天门将至百运开，拂尘轻笑问仙来。
何须扶我青云志，我自凌云至山巅。

**Comment by xwal** [comment]
## 2025-08-07
The Path of Balance - 平衡之道
生活和工作的平衡，收入和支出的平衡，有些是自动平衡，有些是手动平衡，合乎天道。
